### PR TITLE
Chore: Migrate containers/Toast functional component and hooks

### PR DESCRIPTION
**Pull request by samay-kothari**
## Proposed changes

Migrate the containers/Toast from class component to functional component and using hooks. Also removed the interface and `withTheme` HOC, instead using `useTheme` hook. 
## Issue(s)	

Closes https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/4157

## How to test or reproduce

To test the changes:
1. Go to a chat and long hold on a message.
2. Select the copy button from the message actions.
3. You will be able to see the `Copied to Clipboard!` toast.
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by samay-kothari** [comment]
@gerzonc Please have a look

File: app/containers/Toast.tsx [hunk 1]
```diff
@@ -25,50 +25,32 @@ interface IToastProps {
...
-		}
-	}
-
-	getToastRef = (toast: EasyToast | null) => (this.toast = toast);
-
-	showToast = ({ message }: { message: string }) => {
-		if (this.toast && this.toast.show) {
-			this.toast.show(message, 1000);
+const Toast = ({ theme }: IToastProps) => {
+	let listner: Function;
```
Comment (gerzonc) [inline app/containers/Toast.tsx:29]:
```suggestion
	let listener: Function;
```

File: app/containers/Toast.tsx [hunk 2]
```diff
@@ -25,50 +25,32 @@ interface IToastProps {
...
+	let toast: EasyToast | null | undefined;
+	useEffect(() => {
+		listner = EventEmitter.addEventListener(LISTENER, showToast);
```
Comment (gerzonc) [inline app/containers/Toast.tsx:32]:
```suggestion
		listener = EventEmitter.addEventListener(LISTENER, showToast);
```

File: app/containers/Toast.tsx [hunk 3]
```diff
@@ -25,50 +25,32 @@ interface IToastProps {
...
-			EventEmitter.removeListener(LISTENER, this.listener);
-		}
-	}
-
-	getToastRef = (toast: EasyToast | null) => (this.toast = toast);
-
-	showToast = ({ message }: { message: string }) => {
-		if (this.toast && this.toast.show) {
-			this.toast.show(message, 1000);
+const Toast = ({ theme }: IToastProps) => {
```
Comment (gerzonc) [inline app/containers/Toast.tsx:28]: If you remove the `withTheme` HOC and use the `useTheme` hook you won't need this, you could use `colors`
```ts
const { colors } = useTheme();
```

File: app/containers/Toast.tsx [hunk 4]
```diff
@@ -25,50 +25,32 @@ interface IToastProps {
...
-			/>
-		);
-	}
-}
+	return (
+		<EasyToast
+			ref={getToastRef}
+			// @ts-ignore
+			position='center'
+			style={[styles.toast, { backgroundColor: themes[theme!].toastBackground }]}
```
Comment (gerzonc) [inline app/containers/Toast.tsx:49]: By using `colors` from the `useTheme` hook, you can change `themes[theme!].toastBackground` with `colors.toastBackground`

File: app/containers/Toast.tsx [hunk 5]
```diff
@@ -25,50 +25,32 @@ interface IToastProps {
...
+			textStyle={[styles.text, { color: themes[theme!].buttonText }]}
```
Comment (gerzonc) [inline app/containers/Toast.tsx:50]: Same here

File: app/containers/Toast.tsx [hunk 6]
```diff
@@ -25,50 +25,32 @@ interface IToastProps {
 	theme?: TSupportedThemes;
```
Comment (gerzonc) [inline app/containers/Toast.tsx:25]: Won't be needed anymore

**Review by gerzonc — CHANGES_REQUESTED** [review]

**Comment by samay-kothari** [comment]
@gerzonc I have made the requested changes, please review.

**Comment by samay-kothari** [comment]
https://user-images.githubusercontent.com/55805094/165775320-e11f4acd-b0df-4c1f-abc4-8af69bae31a6.mp4

Working as expected after migration

**Review by gerzonc — APPROVED** [review]
LGTM

**Comment by GleidsonDaniel** [comment]
Some points:
When using the ref and listener as let you will cause numerous re-renders, as it will always be initialized again with each render.
Another thing is about this useEffect without [ ], you really want it to run every re-render, shouldn't you just initialize it once?

**Pull request closed without merging**